packages/lesswrong/server/collections/bookmarks/mutations.ts

```diff
@@ -0,0 +1,100 @@
+import gql from 'graphql-tag';
+import { Bookmarks } from './collection';
+import { userCanDo } from "@/lib/vulcan-users/permissions";
+import { logFieldChanges } from "@/server/fieldChanges";
+
+export const bookmarkGqlTypeDefs = gql`
+  input ToggleBookmarkInput {
+    documentId: String!
+    collectionName: String!
+  }
+
+  type ToggleBookmarkOutput {
+    data: Bookmark
+  }
+
+  extend type Mutation {
+    toggleBookmark(input: ToggleBookmarkInput!): ToggleBookmarkOutput
+  }
+`;
+
+interface ToggleBookmarkInput {
+  documentId: string;
+  collectionName: "Posts" | "Comments";
+}
+
+async function toggleBookmarkResolver(root: void, { input }: { input: ToggleBookmarkInput }, context: ResolverContext): Promise<{ data: DbBookmark | null }> {
+  const { documentId, collectionName } = input;
+  const { currentUser } = context;
+
+  if (!currentUser || !userCanDo(currentUser, 'bookmarks.toggle')) {
```

Comment:
'bookmarks.toggle' isn't a permission we have and we shouldn't be using the old vulcan permissions anyways.  Is there any restriction on who can have bookmarks beyond logged in users?

Reply:
nope, removed